### ISSUE TITLE
Convert Core Resource Types to Prototypes

### ISSUE THREAD

**Issue by scottietremendous** [concourse/concourse#6602]
#### TLDR

- Putting the final bow on Prototypes by converting the current core resource types into prototypes. 

#### Problem

- The current Concourse deployment can be quite heavy as it comes pre-packed with 'core' resource types such as `git`
and `s3.
- Some users may not want/need the core resource types

#### Proposed Solution

- Convert these core resources into prototypes.

#### Outcomes

- A much thinner and more flexible Concourse release. 

#### Current Status

- Awaiting the implementation of [prototypes](https://github.com/concourse/concourse/issues/5719)

#### References

- [Prototypes RFC](https://github.com/concourse/rfcs/pull/37)
- [Resource Prototypes RFC](https://github.com/concourse/rfcs/pull/38)
